### DISCUSSION TITLE
Will the custom code migrate automatically with the Sirius Web API?

### DISCUSSION THREAD

**Discussion by lak-proddev** [eclipse-sirius/sirius-components#3110] [Q&A]
Will the custom code migrate automatically with the Sirius Web API?

We have a Sirius-based Eclipse RCP application and trying to move to the cloud-based application.  It has a Sirius graphical editor and other custom Eclipse UI elements. Like Properties view, Wizards, dialogs, and menus. Which extends the SWT or JFace API.

Can I convert existing custom UI components automatically with the Sirius Web bridge API?  If not, what are the alternative solutions?

Thanks in advance

**Comment by sbegaudeau** [comment]
Hi, existing custom user interface code leveraging SWT or JFace would have to be migrated manually to React, TypeScript and CSS.

**Reply by lak-proddev** [reply]
Okay, Thank you very much for the quick reply.
